index.ts

```diff
@@ -32,6 +32,24 @@ export function onInterrupt(handler: Handler): void {
 }
 
 /**
+ * Waits for all promises to settle, then rejects with the first error, if any.
+ */
+export async function promiseAllStrict(
+      promises: Promise<any>[]): Promise<void> {
+  let error: any = null;
```

Comment:
I think this could be a little simpler by converting all promises to resolve with their errors:

```javascript
let error = await Promise.all(promises.map((p) => p.then((r) => null, (e) => e)))
    .find((e) => e != null);
if (error) throw error;
```